verifier.go

```diff
@@ -46,6 +46,9 @@ func NewVerifier(alg Algorithm, key crypto.PublicKey) (Verifier, error) {
 		if !ok {
 			return nil, fmt.Errorf("%v: %w", alg, ErrAlgorithmMismatch)
 		}
+		if !vk.Curve.IsOnCurve(vk.X, vk.Y) {
```

Comment:
Checking only the `IsOnCurve` is OK for `go 1.18` or above. But not for go `1.17` (see [CVE-2022-23806](https://github.com/advisories/GHSA-8c83-vp4v-h7fq)).

Details see https://github.com/golang/go/issues/50974.

Reply:
I didn't know, thanks for pointing that out.